src/Npgsql/NpgsqlCommand.cs

```diff
@@ -831,31 +885,43 @@ await connector.DescribeMessage
                 }
 
                 var statement = _statements[i];
-                var pStatement = statement.PreparedStatement;
-
-                // A statement may be already prepared, already in preparation (i.e. same statement twice
-                // in the same command), or we can't prepare (overloaded SQL)
-                if (pStatement?.State != PreparedState.ToBePrepared)
-                    continue;
+                var statementName = string.Empty;
+                var preSpecifiedInputParameters = new List<NpgsqlParameter>();
```

Comment:
Let's have a static readonly empty list rather than reallocating on each call.

Reply:
Assuming the goal of your comment was performance, I created an overload of `ParseMessage.Populate()` without the `inputParameters` and `typeMapper` parameters which I think makes more sense from a performance perspective.